messaginginapp/src/main/java/io/customer/messaginginapp/gist/presentation/engine/EngineWebView.kt

```diff
@@ -77,83 +75,75 @@ internal class EngineWebView @JvmOverloads constructor(
     @SuppressLint("SetJavaScriptEnabled")
     fun setup(configuration: EngineWebConfiguration) {
         setupTimeout()
-        val jsonString = Gson().toJson(configuration)
-        encodeToBase64(jsonString)?.let { options ->
-            elapsedTimer.start("Engine render for message: ${configuration.messageId}")
-            val messageUrl =
-                "${state.environment.getGistRendererUrl()}/index.html?options=$options"
-            logger.debug("Rendering message with URL: $messageUrl")
-            webView?.let {
-                it.loadUrl(messageUrl)
-                it.settings.javaScriptEnabled = true
-                it.settings.allowFileAccess = true
-                it.settings.allowContentAccess = true
-                it.settings.domStorageEnabled = true
-                it.settings.textZoom = 100
-                it.setBackgroundColor(Color.TRANSPARENT)
-
-                findViewTreeLifecycleOwner()?.lifecycle?.addObserver(this) ?: run {
-                    logger.error("Lifecycle owner not found, attaching interface to WebView manually")
-                    engineWebViewInterface.attach(webView = it)
-                }
+        elapsedTimer.start("Engine render for message: ${configuration.messageId}")
+        val messageData = mapOf("options" to configuration)
+        val jsonString = Gson().toJson(messageData)
+        val messageUrl =
+            "${state.environment.getGistRendererUrl()}/index.html"
+        logger.debug("Rendering message with URL: $messageUrl")
+        webView?.let {
+            it.settings.javaScriptEnabled = true
+            it.settings.allowFileAccess = true
+            it.settings.allowContentAccess = true
+            it.settings.domStorageEnabled = true
+            it.settings.textZoom = 100
+            it.setBackgroundColor(Color.TRANSPARENT)
+
+            findViewTreeLifecycleOwner()?.lifecycle?.addObserver(this) ?: run {
+                logger.error("Lifecycle owner not found, attaching interface to WebView manually")
+                engineWebViewInterface.attach(webView = it)
+            }
 
-                it.webViewClient = object : WebViewClient() {
-                    override fun onPageFinished(view: WebView, url: String?) {
-                        view.loadUrl("javascript:window.parent.postMessage = function(message) {window.${EngineWebViewInterface.JAVASCRIPT_INTERFACE_NAME}.postMessage(JSON.stringify(message))}")
+            it.webViewClient = object : WebViewClient() {
+                override fun onPageFinished(view: WebView, url: String?) {
+                    val script = """
+                    window.postMessage($jsonString, '*');
+                """.trim()
```

Comment:
I understand this is passing information to Web, but if you have any additional context about the script that could help future readers, it would be great to add that as a comment here.

Also, previously the URL sometimes helped us debug which message we were trying to load among other things. With that now being removed from URL, do you think there's anything additional we should log here to help future debugging, or do you feel the existing improved logs will suffice in most cases?

Reply:
open to ideas, i think the state logs would let us know what message is about to be displayed?